gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -53,6 +55,16 @@ public QueryNode build() {
     // All expressions' attributes from the same source. Will only need one downstream query.
     Optional<String> singleSourceForAllAttributes = ExecutionTreeUtils.getSingleSourceForAllAttributes(executionContext);
 
+    EntitiesRequest entitiesRequest = executionContext.getEntitiesRequest();
+
+    // TODO: If there is a filter on a data source, other than EDS, then the flag is a no-op
+    if (entitiesRequest.getIncludeResultsOutsideTimeRange()) {
```

Comment:
Please add some comments explaining this special casing.
Even for the below if condition too.

Reply:
done